src/server/session.ts

```diff
@@ -1592,16 +1592,15 @@ namespace ts.server {
                             const bakedItem: protocol.NavtoItem = {
                                 name: navItem.name,
                                 kind: navItem.kind,
+                                isCaseSensitive: navItem.isCaseSensitive,
+                                matchKind: navItem.matchKind,
                                 file: navItem.fileName,
                                 start: scriptInfo.positionToLineOffset(navItem.textSpan.start),
                                 end: scriptInfo.positionToLineOffset(textSpanEnd(navItem.textSpan))
                             };
                             if (navItem.kindModifiers && (navItem.kindModifiers !== "")) {
                                 bakedItem.kindModifiers = navItem.kindModifiers;
                             }

```

Comment:
`"none"` wasn't a possible value (detected by changing the type to a string literal union), so in practice `matchKind` would always be set.